builder/penv_setup.py

```
@@ -806,6 +806,57 @@ def _install_esptool_from_tl_install(platform, python_exe, uv_executable, uv_cac
+    """
+    if not has_network:
+        return
+
+    uv_env = None
+    if uv_cache_dir:
+        uv_env = dict(os.environ)
+        uv_env["UV_CACHE_DIR"] = str(uv_cache_dir)
+
+    # Check if pio-lock is already installed
+    try:
+        result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)